tests/e2e/metrics/infrastructure.py

```diff
@@ -0,0 +1,11 @@
+from pathlib import Path
+
+from tests.e2e.utils.infrastructure import BaseInfrastructureV2
+
+
+class MetricsStack(BaseInfrastructureV2):
```

Comment:
My first idea was to use strategy pattern and simply inject Resources class as dependency injection in every test group. This way we would have one core class. Your approach is more clean and easier to comprehend though!